src/org/openstreetmap/josm/data/osm/DefaultNameFormatter.java

```diff
@@ -410,7 +411,12 @@ public Comparator<IRelation<?>> getRelationComparator() {
 
     private static String getRelationTypeName(IRelation<?> relation) {
         // see https://josm.openstreetmap.de/browser/osm/applications/editors/josm/i18n/specialmessages.java
-        String name = trc("Relation type", relation.get("type"));
+        String type = relation.get("type");
+        String name = trc("Relation type", type);
+        String route = trc("Route type", relation.get("route"));
+        if (type != null && type.equals("route") && route != null) {
+            name = route + " " + name;
+        }
```

Comment:
`route` should be defined in the `if` statement and the `type` variable is unnecessary.

The `if` statement should probably be `relation.hasTag("type", "route") && relation.hasKey("route")`.

Reply:
I usually prefer using variables like this for efficiency reasons - if you use `hasKey` and `hasTag` like your propose, you will end up doing multiple map lookups for the same key, while my code needs only one for each key. However, I can see that your proposal will result in more readable code, which is probably more important, so I'll update the code.

Reply:
That is true, but I'd rather see that from a profiling run. If there is a use case where it does have significant performance implications, avoiding the additional lookups would be fine.

But keep in mind that the JVM will do _a lot of_ optimizations, not all of which are obvious.
For example, it may generate compiled code that is _very_ similar to what you wrote.

Generally speaking, readability >>> speed. If there is a hot loop that _you have profiled_, feel free to go for speed. I've done this a few times, and I always explain why.

As an example:
```
When profiling <foo> using all the data from <area>, this was xx% of the CPU cycles and yy% of the memory allocations.
```
Examples of `<foo>`
* Validation
* Layer painting

I usually use "Mesa County, CO" as the `<area>`, since that is where I live.
One thing I will note is that memory allocations are a double whammy. They can cause the gc to perform frequent "stop the world" pauses, which stops _everything_. As an example for where reducing the memory allocations _significantly_ improves performance (as the user sees it), see https://github.com/JOSM/josm/commit/0cd6f20aa15d1d78cf436d8c7c7fe5169e62a243. In that commit, I reduced the total time spent validating a specified area by ~75% and a good chunk of that was from reducing the amount of time spent in GC by effectively removing a bunch of unnecessary memory allocations.

Reply:
You already had me convinced, but thanks for following up :-)